pennylane_lightning_gpu/src/algorithms/ObservablesGPUMPI.hpp

```diff
@@ -0,0 +1,403 @@
+// Copyright 2022-2023 Xanadu Quantum Technologies Inc. and contributors.
```

Comment:
If this file doesn't need any changes to work with the MPI version, we can just use the ObservablesGPU.hpp file.

Reply:
It's quite similar, but `applyInPlace` has a different body and the MPI module has no `SparseHamiltonianGPU`. `applyInPlace` has a different interface, so maybe we could move it to `ObservablesGPU.hpp` is shield it with the `ENABLE_MPI` macro?

Reply:
On second look, I think there should be a signature conflict, so I guess we should not template `applyInPlace` on the `StateVectorXXX` class.